### ISSUE TITLE
Styleguide does not work with `sections`

### ISSUE THREAD

**Issue by sobolevn** [vue-styleguidist/vue-styleguidist#405]
**Current behavior**

**To reproduce**

I am using this configuration:

```js
'components': './client/components/**/*.vue',
```

Change it to this configuration:

```js
'sections': [{
    'name': 'Atoms',
    'sections': [
      { 'components': './client/components/atoms/TButton.vue' },
      { 'components': './client/components/atoms/TInput.vue' }
    ],
  }, {
    'name': 'Molecules',
    'components': [],
  }, {
    'name': 'Organisms',
    'components': [
      './client/components/atoms/TCollapsableMenu.vue',
    ],
  }],
```

And now all components do not show up. Only white screen is shown.

**Expected behavior**

I expect it to work the same way.

**Comment by elevatebart** [comment]
Hey @sobolevn, if you open the console of your browser do you get any errors?

I can see one thing that might be not perfect in this config.

You should not have a section without a name

```js
'sections': [{
    'name': 'Atoms',
    'sections': [
      // a section has no name (GoTtm)
      { 'components': './client/components/atoms/TButton.vue' },
      // this one either
      { 'components': './client/components/atoms/TInput.vue' }
    ],
  }, {
    'name': 'Molecules',
    'components': [],
  }, {
    'name': 'Organisms',
    'components': [
      './client/components/atoms/TCollapsableMenu.vue',
    ],
  }],
```

why not do that ?

```js
'sections': [{
    'name': 'Atoms',
    'components': './client/components/atoms/*.vue' ,
    'ignore': './client/components/atoms/TCollapsableMenu.vue'
  }, {
    'name': 'Molecules',
    'components': [],
  }, {
    'name': 'Organisms',
    'components': [
      './client/components/atoms/TCollapsableMenu.vue',
    ],
  }],
```

I hope this is helpful. I will try to make the error message more explicit.

Thank you for reporting ;)

**Comment by sobolevn** [comment]
Adding name does not change anything. Still the same. I do not have any errors in my browser's console.

**Comment by elevatebart** [comment]
Why do you have a section with nothing ?

**Comment by elevatebart** [comment]
Hey @sobolevn, do you have the `defaultExample` flag set to true?

**Comment by elevatebart** [comment]
@sobolevn, I keep trying my hardest to reproduce your bug, can you help me?

Could you modify [this example](https://github.com/vue-styleguidist/vue-styleguidist/tree/try-failingsections/examples/sections) with enough of your setup so that it fails?

Here is how to proceed:
1. hit fork on top of the repo => will create a repo in your github
1. clone the repo on your local machine
1. checkout the `try-failingsections` branch
1. install and launch the `sections` example (`yarn`, `yarn compile` then `yarn start sections`)
1. change the sections example with some of your config until it breaks like your repo
1. send a link to the repo over here

**Comment by sobolevn** [comment]
@elevatebart my setup is quite different:

```ts
// scripts/styleguide.ts

/**
 * Styleguide runner.
 *
 * Use `STYLEGUIDE_ENV` to switch between serve / build modes.
 * 1. `STYLEGUIDE_ENV === 'development'` starts server
 * 2. Any other `STYLEGUIDE_ENV` builds static assets
 */

const { join } = require('path')

const { Nuxt, Builder } = require('nuxt')
const styleguidist = require('vue-styleguidist')

// eslint-disable-next-line @typescript-eslint/no-var-requires
const config = require('../nuxt.config')

const nuxt = new Nuxt({ ...config, 'mode': 'spa' })
const customBuilder = new Builder(nuxt)
const webpackConfig = customBuilder.bundleBuilder.getWebpackConfig('Client')

webpackConfig.plugins.forEach((plugin): void => {
  // This is an ugly hack to place unused `./server` directory
  // to some unused place, should be improved later:
  if (plugin.options && plugin.options.filename) {
    plugin.options.filename = plugin.options.filename.replace(
      '../',
      '../styleguide/',
    )
  }
})

// TODO: enhance styleguide with Vuex, Nuxt's modules, etc
// See nuxtjs example at:
// https://github.com/vue-styleguidist/vue-styleguidist

// See example of configuration:
// https://github.com/vue-styleguidist/vue-styleguidist/blob/master/examples
const styleguide = styleguidist({
  'components': './client/components/**/*.vue',
  'styleguideDir': 'dist/styleguide',
  // TODO: change layout to support sections
  // https://github.com/vue-styleguidist/vue-styleguidist/issues/405
  'require': [
    join(__dirname, '../client/styleguide/style-override.css'),
    ...config.css,
  ],
  'renderRootJsx': join(
    __dirname,
    '../client/styleguide/styleguide.root.ts',
  ),
  webpackConfig,
})

async function buildNuxt (): Promise<void> {
  // At first, we build Nuxt itself:
  await customBuilder.build()
}

async function buildStyleguide (): Promise<void> {
  await buildNuxt()

  // Then we build the styleguide, see API docs:
  // https://vue-styleguidist.github.io/API.html#methods
  styleguide.build((err): void => {
    /* eslint-disable no-console */
    if (err) {
      console.error(err)
      process.exit(1) // eslint-disable-line unicorn/no-process-exit
    } else {
      console.log('Styleguide is built successfully!')
    }
    /* eslint-enable no-console */
  })
}

async function serveStyleguide (): Promise<void> {
  await buildNuxt()

  styleguide.server((err, settings): void => {
    /* eslint-disable no-console */
    if (err) {
      console.log(err)
    } else {
      const url = `http://${settings.serverHost}:${settings.serverPort}`
      console.log(`Listening at ${url}`)
    }
    /* eslint-enable no-console */
  })
}

if (process.env.STYLEGUIDE_ENV === 'development') {
  serveStyleguide()
} else {
  buildStyleguide()
}
```

I run it as `npm run styleguide`: 
```
"styleguide": "cross-env NODE_ENV=production ts-node scripts/styleguide.ts",
```

The rest of source code is available at https://github.com/wemake-services/wemake-vue-template

**Comment by elevatebart** [comment]
@sobolevn thank you so much !!! that is very helpful.
Might I add, this is a clever way to set it up.

**Comment by elevatebart** [comment]
May I PR on your repo?

**Comment by sobolevn** [comment]
@elevatebart thanks :)

Sure, fill free if that helps.

**Comment by elevatebart** [comment]
for the functions `serveStyleguide` and `buildStyleguide` I have created the binutils member. I forgot to document it in the [API page](https://vue-styleguidist.github.io/docs/API.html). I will fix it soon.
[See the example here](https://github.com/vue-styleguidist/vue-styleguidist/blob/dev/packages/vue-cli-plugin-styleguidist/index.js)

Here is what I would replace last 30 lines with

```js
if (process.env.STYLEGUIDE_ENV === 'development') {
  styleguide.binutils.server(args.open).app
} else {
  styleguide.binutils.build()
}
```

**Comment by elevatebart** [comment]
Can you set [verbose](https://vue-styleguidist.github.io/Configuration.html#verbose) to true and [logger](https://vue-styleguidist.github.io/Configuration.html#logger) as logging everything in the console and share your logs? 
The best format to share it would be a text file.

**Comment by elevatebart** [comment]
Another question, why do you need to build nuxt pages before you display components? Is there some dependency from your components to the nuxt server?

**Comment by sobolevn** [comment]
@elevatebart I am not sure. But it was not working with out it. I would be happy to remove it.
Here's the result I am getting: https://timelessis-git-master.kira-bot.now.sh/styleguide/index.html

**Comment by elevatebart** [comment]
Even if styleguidist is able to demo vue components that NuXT displays, it cannot use the NuXT environment. For instance, NuXT routing will conflict with styleguidist routing. So, as soon as you start having multiple pages in styleguidist, NuXT is going to go... *nuts* and break your style guide.

This might be very well happening here. You probably will prefer a more `normal` style guide.
1. Use `styleguide.config.js` and the standard [CLI](https://vue-styleguidist.github.io/docs/CLI.html)
1. Simplify your `webpackConfig` to the core, use the NuXT example to help you.
1. When it fails, open the browser's console and see what plugins are making your demo fail
1. Mock each plugin in the `renderRoot` to fix examples

I hope I helped a little.
If a plugin is giving you some trouble, just tell me

**Comment by elevatebart** [comment]
You can see in the [example here](https://github.com/vue-styleguidist/vue-styleguidist/blob/dev/examples/nuxtjs/styleguide/styleguide.root.js) how I mocked the NuXT $axios plugin.

**Comment by elevatebart** [comment]
@sobolevn do you still see an error?

**Comment by sobolevn** [comment]
Nope, since I use single section for now.

**Comment by bummzack** [comment]
Got the same issue, but only with `3.13.x` (tried with `3.13.1` and `3.13.2`). Downgrading to `3.12.0` made the issue disappear.

**Comment by elevatebart** [comment]
I delivered 3.13.2 yesterday. Did you try with it?

**Comment by bummzack** [comment]
Yes, I did. Errors are in `3.13.1` and `3.13.2`. I did not try `3.13.0`

**Comment by bummzack** [comment]
Steps to reproduce:

```
git clone https://github.com/vue-styleguidist/vue-styleguidist.git
cd vue-styleguidist/examples/sections
npm install
npm run styleguide
```

Open `http://localhost:6060/` in your browser, open the console and observe error messages like:
> [Vue warn]: Unknown custom element: <Placeholder> - did you register the component correctly? For recursive components, make sure to provide the "name" option.

Also navigate to the "Placeholder" in the styleguide and observe how the component isn't being rendered in the styleguide.

Now stop the styleguide server, run:
```
npm install -D vue-styleguidist@3.12.0
npm run styleguide
```

Open `http://localhost:6060/` in your browser again. Observe how the error messages are gone and 
the "Placeholder" component gets rendered correctly.

I hope that helps narrowing down the issue. It seems to only happen when sections are being used. Without sections, the error didn't show even with the latest versions.

**Comment by elevatebart** [comment]
Thank you !! @bummzack

**Comment by elevatebart** [comment]
There was definitely a big issue here. will be fixed in 3.13.3, 
VSG was only picking up the first section registered.

Sorry about that
